test/ResearcherContract.test.js

```diff
@@ -117,4 +130,40 @@ contract("ResearcherContract", (accounts) => {
       assert.equal(researcherExists, false);
     });
   });
+
+  describe("#addWork", () => {
+    context("when is not a researcher", () => {
+      it("should return error", async () => {
+        await expectRevert(addWork(), "Only allowed to researchers");
+      });
+    });
+
+    context("when is a researcher", () => {
```

Comment:
Podemos pôr o código que prepara os specs dentro de um `beforeEach`

```js
beforeEach( async () => {
  await addResearcher("Researcher A", resea1Address);
  await addWork(resea1Address);
});
```

Reply:
Adicionado

Reply:
Seria dentro pro próprio contexto mesmo. Dentro desse no caso `"when is a researcher"`

Reply:
Esse aqui faz sentido mover o `beforeEach` pra o contexto `"when is a researcher"` já que ele é utilizado apenas nele.

se ficar nos `describe #addWork` vai ser executado pra specs que não necessitam de tal. 

Faz sentido?

Reply:
Faz sentido. Alterei aqui.